File: Samples-Xamarin/ControlActivity.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.Support.V7.App;
using Android.Util;

namespace Samples
{
	[Activity (Label = "@string/title_activity_control")]			
	public class ControlActivity : AppCompatActivity
	{
		internal static ExamplesProvider selectedControl;
		private ExpandableListView expList;
		private static int IndicatorLeft = 40;
		private static int IndicatorRight = 8;

		protected override void OnCreate (Bundle bundle)
		{
			base.OnCreate (bundle);
			this.SetContentView(Resource.Layout.activity_control);

			if(SupportActionBar != null && selectedControl != null) {
				SupportActionBar.Title = selectedControl.ControlName();
			}

			this.expList = (ExpandableListView)this.FindViewById (Resource.Id.expListView);
			MoveIndicatorImage ();

			ExamplesAdapter ea = new ExamplesAdapter(selectedControl.Examples());
			this.expList.SetAdapter(ea);

			for (int i = 0; i < ea.GroupCount; i++ ) {
				expList.ExpandGroup(i);
			}

			this.expList.ChildClick += (object sender, ExpandableListView.ChildClickEventArgs e) => {
				ExampleActivity.selectedExampleFragment = (Android.Support.V4.App.Fragment)ea.GetChild(e.GroupPosition, e.ChildPosition);
				Intent exampleIntent = new Intent(this, typeof(ExampleActivity));
				this.StartActivity(exampleIntent);
				e.Handled = true;
			};
		}

		public override void OnWindowFocusChanged (bool hasFocus)
		{
			base.OnWindowFocusChanged (hasFocus);

			MoveIndicatorImage ();
		}

		private void MoveIndicatorImage() {
			DisplayMetrics metrics = new DisplayMetrics();
			WindowManager.DefaultDisplay.GetMetrics (metrics);

			int width = metrics.WidthPixels;
			int left = (int)(IndicatorLeft * metrics.Density + 0.5f);
			int right = (int)(IndicatorRight * metrics.Density + 0.5f);

			if (Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.JellyBeanMr2) {
				expList.SetIndicatorBoundsRelative (width - left, width - right);
			} else {
				expList.SetIndicatorBounds (width - left, width - right);
			}
		}
	}
}

File: Samples-Java/Samples/src/main/java/activities/ControlActivity.java
package activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.DisplayMetrics;
import android.view.View;
import android.widget.ExpandableListView;

import com.telerik.android.sdk.R;

public class ControlActivity extends AppCompatActivity {

    public static ExamplesProvider selectedControl;
    private ExpandableListView expList;
    private static final int INDICATOR_LEFT = 40;
    private static final int INDICATOR_RIGHT = 8;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_control);

        if(getSupportActionBar() != null && selectedControl != null) {
            getSupportActionBar().setTitle(selectedControl.controlName());
        }

        this.expList = (ExpandableListView)this.findViewById(R.id.expListView);
        moveIndicatorImage();

        final ExamplesAdapter ea = new ExamplesAdapter(selectedControl.examples());
        this.expList.setAdapter(ea);

        for (int i = 0; i < ea.getGroupCount(); i++ ) {
            expList.expandGroup(i);
        }

        this.expList.setOnChildClickListener(new ExpandableListView.OnChildClickListener() {
            @Override
            public boolean onChildClick(ExpandableListView parent, View v, int groupPosition, int childPosition, long id) {
                ExampleActivity.selectedExampleFragment = (android.support.v4.app.Fragment) ea.getChild(groupPosition, childPosition);
                Intent exampleIntent = new Intent(ControlActivity.this, ExampleActivity.class);
                ControlActivity.this.startActivity(exampleIntent);
                return true;
            }
        });
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);

        moveIndicatorImage();
    }

    private void moveIndicatorImage() {
        DisplayMetrics metrics = new DisplayMetrics();
        getWindowManager().getDefaultDisplay().getMetrics(metrics);

        int width = metrics.widthPixels;
        int left = (int) (INDICATOR_LEFT * metrics.density + 0.5f);
        int right = (int) (INDICATOR_RIGHT * metrics.density + 0.5f);
        if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
            expList.setIndicatorBoundsRelative(width - left, width - right);
        } else {
            expList.setIndicatorBounds(width - left, width - right);
        }
    }
}
